cli/runtime/config/legacy_clientconfig_factory_test.go

```diff
@@ -0,0 +1,386 @@
+				KnownServers: []*configapi.Server{
+					{
+						Name: "test-mc",
+						Type: configapi.ManagementClusterServerType,
+						ManagementClusterOpts: &configapi.ManagementClusterServer{
+							Endpoint: "test-endpoint",
+							Context:  "test-context",
+							Path:     "test-path",
+						},
+						DiscoverySources: []configapi.PluginDiscovery{
+							{
+								GCP: &configapi.GCPDiscovery{
+									Name:         "test",
+									Bucket:       "test-bucket",
+									ManifestPath: "test-manifest-path",
+								},
+								ContextType: configapi.CtxTypeTMC,
+							},
+						},
+					},
+				},
+				CurrentServer: "test-mc",
+				KnownContexts: []*configapi.Context{
+					{
+						Name: "test-mc",
+						Type: "k8s",
+						ClusterOpts: &configapi.ClusterServer{
+							Endpoint:            "test-context-endpoint",
+							Path:                "test-context-path",
+							Context:             "test-context",
+							IsManagementCluster: true,
+						},
+						DiscoverySources: []configapi.PluginDiscovery{
+							{
+								GCP: &configapi.GCPDiscovery{
+									Name:         "test",
+									Bucket:       "ctx-test-bucket",
+									ManifestPath: "ctx-test-manifest-path",
+								},
+								ContextType: configapi.CtxTypeTMC,
+							},
+						},
+					},
+				},
+				CurrentContext: map[configapi.ContextType]string{
+					configapi.CtxTypeK8s: "test-mc",
+				},
+				ClientOptions: &configapi.ClientOptions{
+					CLI: &configapi.CLIOptions{
+						Repositories: []configapi.PluginRepository{
+							{
+								GCPPluginRepository: &configapi.GCPPluginRepository{
+									Name:       "test",
+									BucketName: "bucket",
+									RootPath:   "root-path",
+								},
+							},
+						},
+						DiscoverySources: []configapi.PluginDiscovery{
+							{
+								GCP: &configapi.GCPDiscovery{
+									Name:         "test",
+									Bucket:       "ctx-test-bucket",
+									ManifestPath: "ctx-test-manifest-path",
+								},
+								ContextType: configapi.CtxTypeTMC,
+							},
+						},
+						UnstableVersionSelector: configapi.VersionSelectorLevel("unstable-version"),
+						Edition:                 configapi.EditionSelector("test=tkg"),
+						BOMRepo:                 "test-bomrepo",
+						CompatibilityFilePath:   "test-compatibility-file-path",
+					},
+					Features: map[string]configapi.FeatureMap{
+						"global": {
+							"tkr-version-v1alpha3-beta":     "false",
+							"context-aware-cli-for-plugins": "true",
+							"context-target":                "false",
+						},
+						"cluster": {
+							"custom-nameservers":      "false",
+							"dual-stack-ipv4-primary": "false",
+							"dual-stack-ipv6-primary": "false",
+						},
+					},
+					Env: map[string]string{
+						"dev":  "dev",
+						"user": "vmw",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			err := PersistNode(tc.node)
+			assert.NoError(t, err)
+
+			err = StoreClientConfig(tc.cfg)
```

Comment:
nit: feels we should be testing more than whether StoreClientConfig saved _something_ successfully.
Are you relying on other preexisting tests to assert that the StoreClientConfig implementation is persisting the _correct_ data?

Reply:
updated the tests with intergration tests